src/contexts/subRedditContext.js

```diff
@@ -0,0 +1,44 @@
+import React, { createContext, useState, useEffect } from 'react';
```

Comment:
Why did you use context in this situation? From my perspective, it looks unnecessary because all of the logic could be moved to the `subredditForm` component without many changes. I'd use context only if multiple components need to access common variables and to avoid "prop drilling".

Reply:
I used it for a couple of reasons: the central state management makes it easy to subscribe state changes, here the benefit in my case is to avoid multiple api requests as `subreddit` changed, the `SubRedditForm` component thus looks much cleaner. I also assume the app will be growing up, so I want to take it as an opportunity to see what challenges I may or not face. Turns out the testing will be much more complex as involving context, hooks and routes altogether. I might change back to the hook approach as I’m not familiar with testing and it took me too much time.